Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,962 @@
+                                                                        @{N='MinimalTlsVersion';E={$_.MinimalTlsVersion}}
+        }
+    }
+
+    $totalsqlServersWithoutReqMinTLSVersion = ($sqlServersWithoutReqMinTLSVersion | Measure-Object).Count
+     
+    if ($totalsqlServersWithoutReqMinTLSVersion  -eq 0)
+    {
+        Write-Host "No SQL Server(s) found where TLS is less than required TLS version.. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        
+        if($AutoRemediation -and ($sqlServerResources |Measure-Object).Count -gt 0) 
+        {
+            $logFile = "LogFiles\"+ $($TimeStamp) + "\log_" + $($SubscriptionId) +".json"
+            $log =  Get-content -Raw -path $logFile | ConvertFrom-Json
+            foreach($logControl in $log.ControlList){
+                if($logControl.ControlId -eq $controlIds){
+                    $logControl.RemediatedResources=$logRemediatedResources
+                    $logControl.SkippedResources=$logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }	
+
+        return
+    }
+
+    Write-Host "Found [$($totalsqlServersWithoutReqMinTLSVersion)] SQL servers where TLS version is either not set or less than required minimal TLS version." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)	
+     
+    if(-not($AutoRemediation))
+    {
+        Write-Host "`nFollowing SQL Servers are having TLS version either not set or less than required minimal TLS version less than required TLS Version:" -ForegroundColor $([Constants]::MessageType.Info)
+        $colsProperty =     @{Expression={$_.ServerName};Label="Server Name";Width=10;Alignment="left"},
+                            @{Expression={$_.ResourceGroupName};Label="Resource Group";Width=10;Alignment="left"},
+                            @{Expression={$_.Location};Label="Location";Width=7;Alignment="left"},
+                            @{Expression={$_.ServerVersion};Label="Server Version";Width=7;Alignment="left"},
+                            @{Expression={$_.minimalTlsVersion};Label="Minimal TLS Version";Width=7;Alignment="left"}
+
+        $sqlServersWithoutReqMinTLSVersion | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetSQLServerMinReqTLSVersion"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "`n[Step 4 of 5] Backing up SQL Server(s) details..."
```

Comment:
In steps, we are not adding '...' in suffix as per standard

Reply:
Removed ... from steps